core.indexes.multi.MultiIndex.from_tuples might have a wrong signature

**Describe the bug**
The `tuples` argument type of core.indexes.multi.MultiIndex.from_tuples might be `Iterables`, not `Sequence`. 

Comparison of the function signature in pandas-stub:
https://github.com/pandas-dev/pandas-stubs/blob/e28bf601e96aea010b803d2984251b2b70dca1c7/pandas-stubs/core/indexes/multi.pyi#L49
and pandas:
https://github.com/pandas-dev/pandas/blob/513e78760a8faee6723908ca23ea83dcabc5dd77/pandas/core/indexes/multi.py#L521

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.

```
import pandas

pandas.MultiIndex.from_tuples(zip(['a'], ['b']))
```

2. Indicate which type checker you are using (`mypy` or  `pyright`).
`mypy`

3. Show the error message received from that type checker while checking your example.
```
error: Argument 1 to "from_tuples" of "MultiIndex" has incompatible type "zip[tuple[str, str]]"; expected "Sequence[tuple[Hashable, ...]]"  [arg-type]
```


**Please complete the following information:**
 - OS: MacOS
 - OS Version: Sequoia 15.3
 - python version: Python 3.10
 - version of type checker: mypy 1.15.0
 - version of installed `pandas-stubs`: 2.2.3.250308

--- a/pandas-stubs/core/indexes/multi.pyi
+++ b/pandas-stubs/core/indexes/multi.pyi
@@ -1,6 +1,7 @@
 from collections.abc import (
     Callable,
     Hashable,
+    Iterable,
     Sequence,
 )
 from typing import (
@@ -46,7 +47,7 @@ class MultiIndex(Index[Any]):
     @classmethod
     def from_tuples(
         cls,
-        tuples: Sequence[tuple[Hashable, ...]],
+        tuples: Iterable[tuple[Hashable, ...]],
         sortorder: int | None = ...,
         names: SequenceNotStr[Hashable] = ...,
     ) -> Self: ...

--- a/tests/test_indexes.py
+++ b/tests/test_indexes.py
@@ -95,6 +95,10 @@ def test_multiindex_constructors() -> None:
         pd.MultiIndex,
     )
     check(
+        assert_type(pd.MultiIndex.from_tuples(zip([1, 2], [3, 4])), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    check(
         assert_type(pd.MultiIndex.from_tuples([(1, 3), (2, 4)]), pd.MultiIndex),
         pd.MultiIndex,
     )

